src/redux/gas.ts

```diff
@@ -555,12 +555,8 @@ export const gasPricesStartPolling =
                   // the basefee at the time we fork mainnet during our hardhat tests
                   let baseFee = baseFeePerGas;
                   if (chainId === ChainId.mainnet && IS_TESTING === 'true') {
-                    const providerUrl = (
-                      web3Provider ||
-                      ({} as {
-                        connection: { url: string };
-                      })
-                    )?.connection?.url;
+                    const provider = getProvider({ chainId: ChainId.mainnet });
+                    const providerUrl = provider?.connection?.url;
                     if (isHardHat(providerUrl)) {
```

Comment:
we could replace this with the `connectedToHardhat` directly instead of dealing with `providerUrl` and the `isHardHat` check?

Reply:
yep pushing

Reply:
i went ahead and also removed all instances of isHardhat